server/src/main/java/io/deephaven/server/session/SessionService.java

```diff
@@ -85,38 +85,41 @@ public StatusRuntimeException transform(final Throwable err) {
                 }
                 return sre;
             } else if (err instanceof InterruptedException) {
-                return securelyWrapError(log, err, Code.UNAVAILABLE);
+                return securelyWrapError(err, Code.UNAVAILABLE);
             } else {
-                return securelyWrapError(log, err, Code.INVALID_ARGUMENT);
+                return securelyWrapError(err, Code.INVALID_ARGUMENT);
             }
         }
 
-        private synchronized StatusRuntimeException securelyWrapError(
-                final Logger log,
-                final Throwable err,
-                final Code statusCode) {
+        private StatusRuntimeException securelyWrapError(@NotNull final Throwable err, final Code statusCode) {
             UUID errorId;
-            Throwable curr = err;
-            int currDepth = 0;
-            boolean needToAdd = false;
-            do {
-                errorId = idCache.getIfPresent(curr);
-                needToAdd |= errorId == null;
-            } while (errorId == null && ++currDepth < MAX_STACK_TRACE_CAUSAL_DEPTH && (curr = curr.getCause()) != null);
+            final boolean shouldLog;
+
+            synchronized (idCache) {
+                errorId = idCache.getIfPresent(err);
+                shouldLog = errorId == null;
+
+                int currDepth = 0;
+                for (Throwable causeToCheck = err.getCause(); errorId == null
```

Comment:
Spotless has done terrible things to these two for loops. It might be worth `// @formatter:off` and `// @formatter:on`.